src/xdsl/dialects/affine.py

```diff
@@ -36,3 +42,208 @@ def store(self, value: Union[Operation, SSAValue],
             SSAValue.get(i),
             SSAValue.get(j)
         ], [], {})
+
+class AffineOperation(Enum):
+    add = '+'
+    mul = '*'
+    ceildiv = '/'
+    floordiv = '//'
+    mod = '%'
+    neg = '-'
+
+@dataclass
+class AffineExpr:
+    def __add__(self, other: int):
+        return self + AffineConstantExpr(other)
+    def __add__(self, otherExpr: AffineExpr):
```

Comment:
How does operator overloading works in Python?
Does this works because of the dataclass?